esp/src/src-react/components/SourceEditor.tsx

```diff
@@ -94,6 +120,36 @@ export const XMLSourceEditor: React.FunctionComponent<XMLSourceEditorProps> = ({
     return <SourceEditor text={text} readonly={readonly} mode="xml"></SourceEditor>;
 };
 
+interface JSONSourceEditorProps {
+    json?: object;
+    readonly?: boolean;
+    onChange?: (obj: object) => void;
+}
+
+export const JSONSourceEditor: React.FunctionComponent<JSONSourceEditorProps> = ({
+    json,
+    readonly = false,
+    onChange = (obj: object) => { }
+}) => {
+
+    const text = React.useMemo(() => {
+        try {
+            return JSON.stringify(json, undefined, "    ");
```

Comment:
stringify will also accept an integer for the number of spaces to insert in the result

Reply:
Changed as suggested...